### PR TITLE
Create variable MSOURCE_TRAP

### PR DESCRIPTION
**Pull request by kphoenix137**
Creates a variable MSOURCE_TRAP defined as -1 to use for checks with _misource.

### CODE REVIEW

File: Source/missiles.cpp [hunk 1]
```diff
@@ -1301,7 +1302,7 @@ void AddFireRune(Missile &missile, Point dst, Direction /*midir*/)
 void AddLightningRune(Missile &missile, Point dst, Direction /*midir*/)
 {
 	int id = missile._misource;
-	int lvl = (id > -1) ? Players[id]._pLevel : 0;
+	int lvl = (id > MSOURCE_TRAP) ? Players[id]._pLevel : 0;
```
Comment (AJenbo) [inline Source/missiles.cpp:1305]:
```suggestion
	int lvl = id != MSOURCE_TRAP ? Players[id]._pLevel : 0;
```

File: Source/missiles.cpp [hunk 2]
```diff
@@ -30,6 +30,7 @@ int AvailableMissiles[MAXMISSILES];
 Missile Missiles[MAXMISSILES];
 int ActiveMissileCount;
 bool MissilePreFlag;
+constexpr int MSOURCE_TRAP = -1;
```
Comment (AJenbo) [inline Source/missiles.cpp:33]: Maybe `MSOURCE_OBJECT` or `MSOURCE_WORLD` would be better names like you suggested.

But also this should be used where the value is being set in the first place, not just for the check.

Comment (kphoenix137) [reply Source/missiles.cpp]: Got the first part. Not sure what you mean for the second part. What am I missing?

**Comment by AJenbo** [comment]
Good clean up, but I liked @qndel take on it better, thanks to both of you :)

**Pull request closed without merging**